### PR TITLE
fix(docs): remove exposed secrets in provider snippets

### PR DESCRIPTION
**Pull request by Maed223**
Closes: #2576

### CODE REVIEW

File: examples/csharp/documentation/ProviderClassesStack.cs [hunk 1]
```diff
@@ -36,10 +36,24 @@ public ProviderClassesStack(Construct scope, string name) : base(scope, name)
...
+                Description = "dnsimple account",
+                Sensitive = true
+            });
+
             new DnsimpleProvider(this, "dnsimple", new DnsimpleProviderConfig
             {
-                Token = Token.AsString(Environment.GetEnvironmentVariable("DNSIMPLE_TOKEN")),
-                Account = Token.AsString(Environment.GetEnvironmentVariable("DNSIMPLE_ACCOUNT"))
+                Token = Token.AsString(dnsimpleToken.StringValue),
+                Account = Token.AsString(dnsimpleAccount.StringValue)
```
Comment (ansgarm) [inline examples/csharp/documentation/ProviderClassesStack.cs:56]: I think the `Token.AsString()` bit can be removed, as `.StringValue` is doing that internally already :)

File: examples/java/documentation/src/main/java/com/mycompany/app/providers/MainImportClasses.java [hunk 2]
```diff
@@ -30,9 +31,23 @@ public MainImportClasses(Construct scope, String id){
...
+                .description("dnsimple account")
+                .sensitive(true)
+                .build()
+        );
+
         new DnsimpleProvider(this, "dnsimple", DnsimpleProviderConfig.builder()
-                .token(Token.asString(System.getenv("DNSIMPLE_TOKEN")))
-                .account(Token.asString(System.getenv("DNSIMPLE_ACCOUNT")))
+                .token(Token.asString(dnsimpleToken.stringValue()))
+                .account(Token.asString(dnsimpleAccount.stringValue()))
```
Comment (ansgarm) [inline examples/java/documentation/src/main/java/com/mycompany/app/providers/MainImportClasses.java:50]: same as above, let's remove `Token.asString()`

File: examples/python/documentation/providers.py [hunk 3]
```diff
@@ -47,9 +47,21 @@ def __init__(self, scope: Construct, id: str):
...
+            type = "string",
+            description = "dnsimple account",
+            sensitive = True
+        )
+
         DnsimpleProvider(self, "dnsimple",
-            token = Token.as_string(os.getenv("DNSIMPLE_TOKEN")),
-            account = Token.as_string(os.getenv("DNSIMPLE_ACCOUNT")),
+            token = Token.as_string(dnsimpleToken.string_value),
+            account = Token.as_string(dnsimpleAccount.string_value)
```
Comment (ansgarm) [inline examples/python/documentation/providers.py:64]: same as above, let's remove `Token.as_string()`

File: examples/typescript/documentation/providers.ts [hunk 4]
```diff
@@ -30,9 +30,21 @@ export class ProvidersStack extends TerraformStack {
...
+      type: "string",
+      description: "dnsimple account",
+      sensitive: true,
+    });
+
     new DnsimpleProvider(this, "dnsimple", {
-      token: Token.asString(process.env.DNSIMPLE_TOKEN),
-      account: Token.asString(process.env.DNSIMPLE_ACCOUNT),
+      token: Token.asString(dnsimpleToken.stringValue),
+      account: Token.asString(dnsimpleAccount.stringValue),
```
Comment (ansgarm) [inline examples/typescript/documentation/providers.ts:47]: aaaaand, one last one: Let's remove `Token.asString()` 😄

**Review by ansgarm — COMMENTED** [review]
I know this is still a draft, but I figured, I'd give it a first pass already 😇
Looking good mostly, just some tiny things then it's good to go!
(I pushed a commit fixing the Go example as it didn't seem to compile)

File: examples/csharp/documentation/ProviderClassesStack.cs (hunk 1 above)
Comment (Maed223) [reply examples/csharp/documentation/ProviderClassesStack.cs:56]: 👍

File: examples/python/documentation/providers.py (hunk 3 above)
Comment (Maed223) [reply examples/python/documentation/providers.py:64]: 👍

File: examples/typescript/documentation/providers.ts (hunk 4 above)
Comment (Maed223) [reply examples/typescript/documentation/providers.ts:47]: 👍

**Review by ansgarm — APPROVED** [review]

**Pull request merged**